test/e2e/apps/job.go

```diff
@@ -111,18 +110,20 @@ var _ = SIGDescribe("Job", func() {
 
 	// Pods sometimes fail, but eventually succeed, after pod restarts
 	ginkgo.It("should run a job to completion when tasks sometimes fail and are not locally restarted", func() {
+		// One failure, then a success, no local restarts.
+		// We can't use the random failure approach, because JobController
+		// will throttle frequently failing Pods of a given Job, ramping
+		// up to 6 minutes between restarts, making test timeout due to
+		// successive failures.
+		// Instead, we force the Job's Pods to be scheduled to a single Node
+		// and use a hostPath volume to persist data across new Pods.
+		ginkgo.By("Looking for a node to schedule job pod")
+		node, err := e2enode.GetRandomReadySchedulableNode(f.ClientSet)
+		framework.ExpectNoError(err)
+
 		ginkgo.By("Creating a job")
-		// 50% chance of container success, local restarts.
-		// Can't use the failOnce approach because that relies
-		// on an emptyDir, which is not preserved across new pods.
-		// Worst case analysis: 15 failures, each taking 1 minute to
-		// run due to some slowness, 1 in 2^15 chance of happening,
-		// causing test flake.  Should be very rare.
-		// With the introduction of backoff limit and high failure rate this
-		// is hitting its timeout, the 3 is a reasonable that should make this
-		// test less flaky, for now.
-		job := jobutil.NewTestJob("randomlySucceedOrFail", "rand-non-local", v1.RestartPolicyNever, parallelism, 3, nil, 999)
-		job, err := jobutil.CreateJob(f.ClientSet, f.Namespace.Name, job)
+		job := jobutil.NewTestJob("failOnce", "fail-once-non-local", v1.RestartPolicyNever, parallelism, completions, nil, backoffLimit, node.Name)
```

Comment:
Is it intentional that we are changing the value of `completions` and `backoffLimit` values here? My guess is that the change is intentional... from my reading of the comment, it seemed like we had to set some custom values for these because we couldn't rely on the `failOnce` approach. But I just want to double check.

Reply:
I checked the commit history of the test and found it was decreased to 3 to make is less flaky as the default completions 4 was harder to reach and it expected to fail many times when it depended on random failure approach:
https://github.com/kubernetes/kubernetes/commit/90e7b5ff4fa5dc481ac4367d743da0d9d3fd3bcb
https://github.com/kubernetes/kubernetes/commit/9991a88870a8b6674e46e64e5d91f5bc9ea720b8
So I think as it's going to be stable after switching to `failOnce` approach, better to keep unified arguments for the similar two cases. What do you think?